docs/guides/version-compatibility.rst

```diff
@@ -0,0 +1,44 @@
+=====================
+Version Compatibility
+=====================
+
+This guide covers the details behind the warning::
+
+    VersionMismatch: Attempting to read a workflow definition generated with a different version of Orquestra Workflow SDK. Please consider re-running your workflow or installing 'orquestra-sdk==...'.
+
+
+Client-Server Coupling
+======================
+
+When you're running workflows remotely, you have a local version of Orquestra Workflow SDK on your local machine or the Jupyter environment in Orquestra Portal.
+An Intermediate Representation (IR) is generated from your ``@workflow``-decorated function and sent over the wire to the remote runtime.
+
+The remote runtime also comes with its own installation of the Workflow SDK to interpret the workflow IR, execute tasks, and store results.
+The Workflow SDK version installed on the server side is selected to match the client's one.
+
+Finally, you're likely to interact with the workflow run to query its status, peek into the graph of task runs, or access the data they produced.
+This interaction can happen a long time after the workflow was originally executed; the workflow data lives longer than the Workflow SDK code that produced it.
```

Comment:
I tried to epxand on some points, idk if it worked.

What I was trying to get across is that this version coupling doesn't only exist for remote workflows, it exists locally too. The only special behaviour is the matching of the SDK version.

```suggestion
Workflow Version Coupling
=========================

When you write a workflow, the Orquestra Workflow SDK generates an Intermediate Representation (IR) from your ``@workflow``-decorated function. When you "run" the workflow, this IR is sent to a runtime for execution. Inside the IR, we store the version of the Workflow SDK that generated it.

When executing the workflow remotely, the remote runtime reads the Workflow SDK version from the IR and uses the same version to interpret the IR, execute tasks, and store results.

Once a workflow run has been started, you will query its status, peek into its graph of task runs, or access the data it produced.

Sometimes these interactions can happen a long time after the workflow was originally executed; the workflow data lives longer than the Workflow SDK version that produced it.

In the meantime, the internals of the Workflow SDK may have changed resulting in unexpected behavior.
```

Reply:
Yeah, makes more sense to me. Added in 5695438b85